openHAB3 sitemap creation failure: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean

Hey,

i'm just testing arount with openHAB3 and run into the following problem.

I've created an sitemap with the new UI-Editor (which is pretty smart). The code sections says that everything is okay.
But when requesting the sitemaps i'm getting the following error within the server log:

```
2020-11-30 18:15:54.881 [ERROR] [omponents.UIComponentSitemapProvider] - Cannot build sitemap page_59d08b12be
java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
        at org.openhab.core.model.sitemap.sitemap.impl.SliderImpl.eSet(SliderImpl.java:320) ~[bundleFile:?]
        at org.openhab.core.ui.internal.components.UIComponentSitemapProvider.setWidgetPropertyFromComponentConfig(UIComponentSitemapProvider.java:286) ~[bundleFile:?]
        at org.openhab.core.ui.internal.components.UIComponentSitemapProvider.buildWidget(UIComponentSitemapProvider.java:214) ~[bundleFile:?]
        at org.openhab.core.ui.internal.components.UIComponentSitemapProvider.buildWidget(UIComponentSitemapProvider.java:262) ~[bundleFile:?]
        at org.openhab.core.ui.internal.components.UIComponentSitemapProvider.buildSitemap(UIComponentSitemapProvider.java:146) ~[bundleFile:?]
        at org.openhab.core.ui.internal.components.UIComponentSitemapProvider.getSitemapNames(UIComponentSitemapProvider.java:105) [bundleFile:?]
        at org.openhab.core.io.rest.sitemap.internal.SitemapResource.getSitemapBeans(SitemapResource.java:413) [bundleFile:?]
        at org.openhab.core.io.rest.sitemap.internal.SitemapResource.getSitemaps(SitemapResource.java:229) [bundleFile:?]
        at jdk.internal.reflect.GeneratedMethodAccessor67.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
```

This is the content of my sitemap (copied from ui code section)

```
sitemap page_59d08b12be label="Zuhause" {
    Frame icon="corridor" label="Flur" item=locationCorridor {
        Slider icon="light" label="Spots Garderobea" item=Garderobe_Helligkeit switchEnabled="true"
    }
}

```

Technical information:

```
runtimeInfo:
  version: 3.0.0.M3
  buildString: Milestone Build
locale: de_DE
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.9
  javaVendor: Azul Systems, Inc.
  javaVendorVersion: Zulu11.43+88-CA
  osName: Linux
  osVersion: 5.4.72-v7l+
  osArchitecture: arm
  availableProcessors: 4
  freeMemory: 47826432
  totalMemory: 127819776
bindings:
  - avmfritz
  - denonmarantz
  - heos
  - network
  - tradfri
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1.5
    prefersColorScheme: light
  isSecureContext: false
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 8
    language: de-DE
    languages:
      - de-DE
      - de
      - en-US
      - en
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
    like Gecko) Chrome/87.0.4280.66 Safari/537.36
timestamp: 2020-11-30T17:22:23.525Z

```

Let me know if you need some further information!

Sönke

--- a/bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java
+++ b/bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java
@@ -12,6 +12,7 @@
  */
 package org.openhab.core.ui.internal.components;
 
+import java.math.BigDecimal;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
@@ -178,7 +179,7 @@ public class UIComponentSitemapProvider implements SitemapProvider, RegistryChan
             case "Video":
                 VideoImpl videoWidget = (VideoImpl) SitemapFactory.eINSTANCE.createVideo();
                 widget = videoWidget;
-                setWidgetPropertyFromComponentConfig(widget, component, "url", SitemapPackage.IMAGE__URL);
+                setWidgetPropertyFromComponentConfig(widget, component, "url", SitemapPackage.VIDEO__URL);
                 setWidgetPropertyFromComponentConfig(widget, component, "encoding", SitemapPackage.VIDEO__ENCODING);
                 break;
             case "Chart":
@@ -282,8 +283,21 @@ public class UIComponentSitemapProvider implements SitemapProvider, RegistryChan
         if (value == null) {
             return;
         }
-        WidgetImpl widgetImpl = (WidgetImpl) widget;
-        widgetImpl.eSet(feature, ConfigUtil.normalizeType(value));
+        try {
+            WidgetImpl widgetImpl = (WidgetImpl) widget;
+            Object normalizedValue = ConfigUtil.normalizeType(value);
+            if (widgetImpl.eGet(feature, false, false) instanceof Integer) {
+                normalizedValue = (normalizedValue instanceof BigDecimal) ? ((BigDecimal) normalizedValue).intValue()
+                        : Integer.valueOf(normalizedValue.toString().split(".")[0]);
+            }
+            if (widgetImpl.eGet(feature, false, false) instanceof Boolean && !(normalizedValue instanceof Boolean)) {
+                normalizedValue = Boolean.valueOf(normalizedValue.toString());
+            }
+            widgetImpl.eSet(feature, normalizedValue);
+        } catch (Exception e) {
+            logger.warn("Cannot set {} parameter for {} widget parameter: {}", configParamName, component.getType(),
+                    e.getMessage());
+        }
     }
 
     private void addWidgetMappings(EList<Mapping> mappings, UIComponent component) {

--- a/bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java
+++ b/bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java
@@ -221,7 +221,6 @@ public class UIComponentSitemapProvider implements SitemapProvider, RegistryChan
                 SelectionImpl selectionWidget = (SelectionImpl) SitemapFactory.eINSTANCE.createSelection();
                 addWidgetMappings(selectionWidget.getMappings(), component);
                 widget = selectionWidget;
-                setWidgetPropertyFromComponentConfig(widget, component, "height", SitemapPackage.WEBVIEW__HEIGHT);
                 break;
             case "List":
                 ListImpl listWidget = (ListImpl) SitemapFactory.eINSTANCE.createList();
@@ -288,9 +287,9 @@ public class UIComponentSitemapProvider implements SitemapProvider, RegistryChan
             Object normalizedValue = ConfigUtil.normalizeType(value);
             if (widgetImpl.eGet(feature, false, false) instanceof Integer) {
                 normalizedValue = (normalizedValue instanceof BigDecimal) ? ((BigDecimal) normalizedValue).intValue()
-                        : Integer.valueOf(normalizedValue.toString().split(".")[0]);
-            }
-            if (widgetImpl.eGet(feature, false, false) instanceof Boolean && !(normalizedValue instanceof Boolean)) {
+                        : Integer.valueOf(normalizedValue.toString());
+            } else if (widgetImpl.eGet(feature, false, false) instanceof Boolean
+                    && !(normalizedValue instanceof Boolean)) {
                 normalizedValue = Boolean.valueOf(normalizedValue.toString());
             }
             widgetImpl.eSet(feature, normalizedValue);

